tensorflow/core/platform/file_system_helper.cc

```diff
@@ -55,30 +55,44 @@ void ForEach(int first, int last, const std::function<void(int)>& f) {
 Status GetMatchingPaths(FileSystem* fs, Env* env, const string& pattern,
                         std::vector<string>* results) {
   results->clear();
-  // Find the fixed prefix by looking for the first wildcard.
-  string fixed_prefix = pattern.substr(0, pattern.find_first_of("*?[\\"));
-  string eval_pattern = pattern;
-  std::vector<string> all_files;
-  string dir(io::Dirname(fixed_prefix));
-  // If dir is empty then we need to fix up fixed_prefix and eval_pattern to
-  // include . as the top level directory.
-  if (dir.empty()) {
-    dir = ".";
-    fixed_prefix = io::JoinPath(dir, fixed_prefix);
-    eval_pattern = io::JoinPath(dir, pattern);
+  if (pattern.empty()) {
+    return Status::OK();
   }
 
+  string eval_pattern = pattern;
+  bool is_directory = pattern[pattern.size() - 1] == '/';
+#ifdef PLATFORM_WINDOWS
+  is_directory = is_directory || pattern[pattern.size() - 1] == '\\';
+#endif
```

Comment:
Why not use `IsDirectory`?

Reply:
@mihaimaruseac Because the `fs->IsDirectory` will check `FileExists` first and here I'm checking a pattern, which is not an existing file.